storyboarding/scripting/compound_commands.rst

```diff
@@ -12,6 +12,405 @@ Loop
+
+On the Background layer we will put a picture of a street that is sidescrolling.
+
+In Pass-state there will be an animation of Aoba running.
+
+In Fail-state Aoba will lie on the ground after having tripped.
+
+
+On triggering `Failing`, Aoba will be tripping.
+
+On triggering `Passing`, Aoba will get up from the ground.
+
+The implementation
+..................
+
+For the sidescrolling street we will take it easy as a start. After a quick google search a 2.5D animation of scrolling buildings turns up.
+
+Splitting that into frames, renaming the individual pictures to use as an animation (I used a script for this because it has about 250 files) and we're ready to go.
+
+.. image:: img/compound_commands/streetscroll.gif
+  :scale: 100%
+  :alt: Scrolling buildings
+  :align: left
+
+Now all we have to do is creating an animation with our existing knowledge:
+
+.. code-block:: c
+  :linenos:
+  :caption: Buildings passing along...
+  :name: StreetScrollCode
+
+  Animation,Background,Centre,"sb\streetscroll\streetscroll.png",320,140,240,150,LoopForever
+   F,0,0,,1
+   S,0,0,,2
+   F,0,90000,,0
+
+And the street is running. Now to the more exciting stuff...
+
+I prepared some animations to use for running, tripping and getting up (actually the hardest part).
+
+.. image:: img/compound_commands/running.gif
+  :scale: 100%
+  :alt: Running character
+  :align: left
+
+.. image:: img/compound_commands/falling.gif
+  :scale: 100%
+  :alt: Falling character
+  :align: left
+
+.. image:: img/compound_commands/gettingUp.gif
+  :scale: 100%
+  :alt: Character getting up.
+  :align: left
+
+Let's start by putting the `Pass`-layer into place. While the gamestate is `Pass` the running-animation is displayed.
+
+.. code-block:: c
+  :linenos:
+  :caption: Character running in Pass-state
+  :name: RunningSimple
+
+  Animation,Background,Centre,"sb\running\running.png",320,240,240,150,LoopForever
+   F,0,0,,1
+
+Simple as that. We have to make another addition for the case that the `Passing` event is triggered.
+
+When this happens, Aoba is supposed to get up first before she starts running again. This means we have to fade the animation out for the process of getting up.
+
+.. code-block:: c
+  :linenos:
+  :caption: Character running in Pass-state with transition
+  :name: RunningComplete
+
+  Animation,Pass,Centre,"sb\running\running.png",320,240,6,150,LoopForever
+   F,0,0,,1
+   T,Passing,0,90000
+    F,0,0,,0
+    F,0,1000,,1
+
+The value of 1000 is arbitrary, it has to be set to the actual duration of getting up.
+
+Now the same is done for the `Fail`-layer with the `Failing` trigger. This time it is a sprite, not an animation as Aoba is just lying down.
+
+.. code-block:: c
+  :linenos:
+  :caption: Character lying on the ground
+  :name: Tripped
+
+  Sprite,Fail,Centre,"sb\falling\fallen.png",280,240
+   F,0,0,,1
+   T,Failing,0,90000
+    F,0,0,,0
+    F,0,2000,,1
+
+All that is left is putting the actual animations for falling and getting up in.
```

Comment:
No need for that hanging preposition.

Reply:
Uh oh, what was a hanging preposition again?
Reconstructed somehow.